encoder.go

```diff
@@ -65,6 +66,7 @@ func (e *CallerEncoder) UnmarshalText(text []byte) error {
 func encodeCaller(c *caller, enc zapcore.PrimitiveArrayEncoder) {
 	// this function can only be called internally so we have full control over it
 	// and can ensure that enc is always of type zapcore.ArrayEncoder
+	c.Function = runtime.FuncForPC(c.PC).Name()
```

Comment:
I don't think the encoder should be responsible for determining the function name, only for encoding it if it is recorded.

Zap will already record Function if `WithCaller(true)` is used: https://github.com/uber-go/zap/blob/a55bdc32f526699c3b4cc51a2cc97e944d02fbbf/logger.go#L374-L382

Reply:
You're right. Thanks for reviewing. Encoder tests manually marshal the EntryCaller, and EntryCaller's Function fields gets filled with zap's unexported check() method, so the function field was empty in tests. I fixed the tests.